### ISSUE TITLE
deprecate opencage_ functions

### ISSUE THREAD

**Issue by dpprdan** [ropensci/opencage#72]
`opencage_forward`, `opencage_reverse` and `opencage_key` need to be deprecated (i.e. give a warning). 

We deleted `opencage_key` prematurely, so I added it again in this branch: https://github.com/dpprdan/opencage/tree/fix/opencage_key

https://devdevguide.netlify.com/evolution.html#functions-deprecate-defunct

move `opencage_format()` to `deprecated.R`

**Comment by dpprdan** [comment]
Having second thoughts about throwing `.Deprecated` warnings every time e.g. `opencage_forward` is called. In fact, if a user calls `opencage_key()` explicitly in a `opencage_forward()` call (i.e. uses the default), there will be two warnings. So geocoding 10 addresses in a skript with the deprecated functions would result in 20 warnings which seems a bit chatty. 

Alternatives:

1. Soft-deprecate a function tidyverse-style, i.e. throw only one warning per (interactive) session. The relevant [code](https://github.com/r-lib/rlang/blob/master/R/compat-lifecycle.R) is not exported from rlang however, and our licences are not 100% compatible (GPL 3 vs GPL >= 2), so that is not an option.
2. Only show a message for now, not a warning. That would still be chatty though.
3. Do not show a message or warning at all, but only rely on the documentation and NEWS for now.
4. Use https://github.com/ropenscilabs/conditionz. Not sure whether it supports showing only one warning per session (and function), though, and if it does, how to do it.

**Comment by jessesadler** [comment]
PR #86 closes this

**Comment by dpprdan** [comment]
🤔 Why did the [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) in the commit message not close the issue?
❗️ "Once the branch is merged into the _default_ branch, the issue will close." (Emphasis mine).
